// @guildeducationinc/recess/components/TableRowCell/TableRowCell.js
/* eslint-disable jsx-a11y/click-events-have-key-events */
import * as React from 'react';
import { cell, currency, text, date, addEllipsis, cellContentPadding } from './index.module.css.js';
import classNames from '../../utilities/classnames';
var renderCell = function (props) {
    if (props.cellRenderer) {
        return props.cellRenderer(props, props.columnKey);
    }
    else {
        return React.createElement("p", { className: cellContentPadding }, props.data);
    }
};
var cssType = function (type) {
    switch (type) {
        case 'currency':
            return currency;
        case 'text':
            return text;
        case 'date':
            return date;
        default:
            return null;
    }
};
export var TableRowCell = function (props) {
    var ellipsis = props.wrapText ? null : addEllipsis;
    var classes = classNames(cell, cssType(props.type), ellipsis);
    return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
    React.createElement("td", { className: classes, "data-testid": '_table-cell', onClick: props.cellClick, style: {
            minWidth: props.minWidth,
            maxWidth: props.maxWidth,
            width: props.width,
            flex: props.flex + " " + props.flex + " " + (props.flex / props.flexBasisMultiplier) * 100 + "%",
            padding: props.cellRenderer && 0,
        } }, renderCell(props)));
};
TableRowCell.defaultProps = {
    type: 'text',
    flex: 1,
    wrapText: false,
};
