// @guildeducationinc/recess/components/StyleGuide/recessColors.js
export var recessColors = {
    teal: {
        base: '#007d8c',
        'darken-25': '#005d69',
        'darken-50': '#003e46',
    },
    blue: {
        base: '#063347',
        'lighten-50': '#37b5ee',
        'lighten-75': '#c2e4f3',
        'lighten-97': '#eff6f8',
    },
    black: {
        base: '#303030',
        'lighten-25': '#636363',
        'lighten-50': '#979797',
        'lighten-70': '#c0c0c0',
        'lighten-87': '#e4e4e4',
        'lighten-90': '#eaeaea',
        'lighten-97': '#f8f8f8',
    },
    white: {
        base: '#ffffff',
    },
    red: {
        base: ' #e00022',
        'darken-25': '#a80019',
        'darken-50': '#700011',
    },
    green: {
        base: '#3dad0e',
    },
    yellow: {
        base: '#ffc500',
    },
};
export var recessColorArray = Object.keys(recessColors).reduce(function (acc, baseColor) {
    Object.keys(recessColors[baseColor]).forEach(function (variation) {
        acc.push({
            colorName: baseColor + "-" + variation,
            sassColor: "$color-" + baseColor + "-" + variation,
        });
    });
    return acc;
}, []);
