// @guildeducationinc/recess/constants/cobranded-enterprise.js
var IMIGX_LINK = 'https://recess-images.guildeducation.com';
export var CO_BRANDED_ENTERPRISE = {
    chipotle_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/chipotle_guild_color.svg',
    chipotle_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/chipotle_guild_gray.svg',
    chipotle_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/chipotle_guild_white.svg',
    davita_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/davita_guild_color.svg',
    davita_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/davita_guild_gray.svg',
    davita_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/davita_guild_white.svg',
    discover_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/discover_guild_color.svg',
    discover_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/discover_guild_gray.svg',
    discover_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/discover_guild_white.svg',
    dps_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/dps_guild_color.svg',
    dps_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/dps_guild_gray.svg',
    dps_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/dps_guild_white.svg',
    hhusa_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/hhusa_guild_color.svg',
    hhusa_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/hhusa_guild_gray.svg',
    hhusa_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/hhusa_guild_white.svg',
    lowes_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/lowes_guild_color.svg',
    lowes_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/lowes_guild_gray.svg',
    lowes_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/lowes_guild_white.svg',
    lyft_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/lyft_guild_color.svg',
    lyft_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/lyft_guild_gray.svg',
    lyft_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/lyft_guild_white.svg',
    m43_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/m43_guild_color.svg',
    m43_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/m43_guild_gray.svg',
    m43_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/m43_guild_white.svg',
    pscu_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/pscu_guild_color.svg',
    pscu_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/pscu_guild_gray.svg',
    pscu_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/pscu_guild_white.svg',
    psl_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/psl_guild_color.svg',
    psl_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/psl_guild_gray.svg',
    psl_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/psl_guild_white.svg',
    sage_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/sage_guild_color.svg',
    sage_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/sage_guild_gray.svg',
    sage_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/sage_guild_white.svg',
    seiu_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/seiu_guild_color.svg',
    seiu_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/seiu_guild_gray.svg',
    seiu_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/seiu_guild_white.svg',
    tacobell_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/tacobell_guild_color.svg',
    tacobell_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/tacobell_guild_gray.svg',
    tacobell_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/tacobell_guild_white.svg',
    walmart_guild_color: IMIGX_LINK + 'enterprise_guild_cobranded_logos/color/walmart_guild_color.svg',
    walmart_guild_gray: IMIGX_LINK + 'enterprise_guild_cobranded_logos/gray/walmart_guild_gray.svg',
    walmart_guild_white: IMIGX_LINK + 'enterprise_guild_cobranded_logos/white/walmart_guild_white.svg',
};
