// @guildeducationinc/recess/constants/colors.js
import { recessColors } from '../components/StyleGuide/recessColors';
export var RecessColors = {
    black: {
        base: recessColors.black.base,
        lighten25: recessColors.black['lighten-25'],
        lighten50: recessColors.black['lighten-50'],
        lighten70: recessColors.black['lighten-70'],
        lighten87: recessColors.black['lighten-87'],
        lighten90: recessColors.black['lighten-90'],
        lighten97: recessColors.black['lighten-97'],
    },
    blue: {
        base: recessColors.blue.base,
        lighten50: recessColors.blue['lighten-50'],
        lighten75: recessColors.blue['lighten-75'],
        lighten97: recessColors.blue['lighten-97'],
    },
    green: {
        base: recessColors.green.base,
    },
    red: {
        base: recessColors.red.base,
        darken25: recessColors.red['darken-25'],
        darken50: recessColors.red['darken-50'],
    },
    teal: {
        base: recessColors.teal.base,
        darken25: recessColors.teal['darken-25'],
        darken50: recessColors.teal['darken-50'],
    },
    yellow: {
        base: recessColors.yellow.base,
    },
};
