// @guildeducationinc/recess/constants/enterprise-partners.js
var IMIGX_LINK = 'https://recess-images.guildeducation.com';
export var ENTERPRISE_PARTNERS = {
    chipotle_color: IMIGX_LINK + 'enterprise_partner_logos/color/chipotle_color.svg',
    chipotle_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/chipotle_gray.svg',
    chipotle_white: IMIGX_LINK + 'enterprise_partner_logos/white/chipotle_white.svg',
    davita_color: IMIGX_LINK + 'enterprise_partner_logos/color/davita_color.svg',
    davita_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/davita_gray.svg',
    davita_white: IMIGX_LINK + 'enterprise_partner_logos/white/davita_white.svg',
    discover_color: IMIGX_LINK + 'enterprise_partner_logos/color/discover_color.svg',
    discover_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/discover_gray.svg',
    discover_white: IMIGX_LINK + 'enterprise_partner_logos/white/discover_white.svg',
    dps_color: IMIGX_LINK + 'enterprise_partner_logos/color/dps_color.svg',
    dps_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/dps_gray.svg',
    dps_white: IMIGX_LINK + 'enterprise_partner_logos/white/dps_white.svg',
    hhusa_color: IMIGX_LINK + 'enterprise_partner_logos/color/hhusa_color.svg',
    hhusa_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/hhusa_gray.svg',
    hhusa_white: IMIGX_LINK + 'enterprise_partner_logos/white/hhusa_white.svg',
    lowes_color: IMIGX_LINK + 'enterprise_partner_logos/color/lowes_color.svg',
    lowes_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/lowes_gray.svg',
    lowes_white: IMIGX_LINK + 'enterprise_partner_logos/white/lowes_white.svg',
    lyft_color: IMIGX_LINK + 'enterprise_partner_logos/color/lyft_color.svg',
    lyft_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/lyft_gray.svg',
    lyft_white: IMIGX_LINK + 'enterprise_partner_logos/white/lyft_white.svg',
    m43_color: IMIGX_LINK + 'enterprise_partner_logos/color/m43_color.svg',
    m43_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/m43_gray.svg',
    m43_white: IMIGX_LINK + 'enterprise_partner_logos/white/m43_white.svg',
    pscu_color: IMIGX_LINK + 'enterprise_partner_logos/color/pscu_color.svg',
    pscu_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/pscu_gray.svg',
    pscu_white: IMIGX_LINK + 'enterprise_partner_logos/white/pscu_white.svg',
    psl_color: IMIGX_LINK + 'enterprise_partner_logos/color/psl_color.svg',
    psl_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/psl_gray.svg',
    psl_white: IMIGX_LINK + 'enterprise_partner_logos/white/psl_white.svg',
    sage_color: IMIGX_LINK + 'enterprise_partner_logos/color/sage_color.svg',
    sage_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/sage_gray.svg',
    sage_white: IMIGX_LINK + 'enterprise_partner_logos/white/sage_white.svg',
    seiu_color: IMIGX_LINK + 'enterprise_partner_logos/color/seiu_color.svg',
    seiu_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/seiu_gray.svg',
    seiu_white: IMIGX_LINK + 'enterprise_partner_logos/white/seiu_white.svg',
    tacobell_color: IMIGX_LINK + 'enterprise_partner_logos/color/tacobell_color.svg',
    tacobell_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/tacobell_gray.svg',
    tacobell_white: IMIGX_LINK + 'enterprise_partner_logos/white/tacobell_white.svg',
    walmart_color: IMIGX_LINK + 'enterprise_partner_logos/color/walmart_color.svg',
    walmart_gray: IMIGX_LINK + 'enterprise_partner_logos/gray/walmart_gray.svg',
    walmart_white: IMIGX_LINK + 'enterprise_partner_logos/white/walmart_white.svg',
};
