// @guildeducationinc/recess/constants/guild-logos.js
var IMGIX_GUILD_LOGO_LINK = 'https://recess-images.guildeducation.com/guild_logos';
export var LOGOS_G_ADMIN = {
    g_admin_color: IMGIX_GUILD_LOGO_LINK + "/g_admin/g_admin_color",
};
export var LOGOS_GUILD = [
    {
        name: 'guild_black',
        path: IMGIX_GUILD_LOGO_LINK + "/guild/guild_black",
    },
    {
        name: 'guild_color',
        path: IMGIX_GUILD_LOGO_LINK + "/guild/guild_color",
    },
    { name: 'guild_gray', path: IMGIX_GUILD_LOGO_LINK + "/guild/guild_gray" },
    {
        name: 'guild_white',
        path: IMGIX_GUILD_LOGO_LINK + "/guild/guild_white",
    },
    {
        name: 'guild_nav',
        path: "https://recess-images.guildeducation.com/enterprise_guild_cobranded_logos/color/standard_size/guild_color",
    },
];
export var LOGOS_G = [
    {
        name: 'g_black',
        path: IMGIX_GUILD_LOGO_LINK + "/g/g_black",
    },
    {
        name: 'g_color',
        path: IMGIX_GUILD_LOGO_LINK + "/g/g_color",
    },
    {
        name: 'g_gray',
        path: IMGIX_GUILD_LOGO_LINK + "/g/g_gray",
    },
    {
        name: 'g_white',
        path: IMGIX_GUILD_LOGO_LINK + "/g/g_white",
    },
];
