// @guildeducationinc/recess/components/Select/index.module.css.js
export const boxShadowSm = 'boxShadow--sm--3hR1m'; 
export const selectPanelWrapper = 'selectPanelWrapper--UJod6'; 
export const boxShadowMd = 'boxShadow--md--1DcfZ'; 
export const boxShadowLg = 'boxShadow--lg--2W3Ok'; 
export const selectFieldWrapper = 'selectFieldWrapper--3XWed'; 
export const usingNonPortalDropdown = 'usingNonPortalDropdown--3L4mm'; 
export const nonPortalDropdown = 'nonPortalDropdown--30g6x'; 
export const nonPortalDropdownNoLabel = 'nonPortalDropdownNoLabel--LDJR8'; 
export const labelWrapper = 'labelWrapper--2U6PD'; 
export const label = 'label--gL0eH'; 
export const disabled = 'disabled--1jlRM'; 
export const invalid = 'invalid--2Ox-n'; 
export const error = 'error--22QmP'; 
export const hidden = 'hidden--1nyYW'; 
export const selectTrigger = 'selectTrigger--2ydH_'; 
export const open = 'open--1sI3S'; 
export const svgArrows = 'svgArrows--2BuJc'; 
export const selectPlaceholder = 'selectPlaceholder--2M8iS'; 
export const listOptions = 'listOptions--2rbeS'; 
export const listOption = 'listOption--3bEqG'; 
export const selectItemSvgIconFill = 'selectItemSvgIconFill--1Ftne'; 
export const focusedListOption = 'focusedListOption--byWmZ'; 
export const optionLabel = 'optionLabel--2Ior4'; 
export const disabledOption = 'disabledOption--3r5hU'; 
export const optionLabelChecked = 'optionLabelChecked--1-mvG';
