// @guildeducationinc/recess/components/Footer/index.module.css.js
export const row = 'row--191VK'; 
export const col = 'col--3IeM4'; 
export const colXs1 = 'col-xs-1--2kZW8'; 
export const colXs2 = 'col-xs-2--1jVs1'; 
export const colXs3 = 'col-xs-3--Pvwso'; 
export const colXs4 = 'col-xs-4--2jrLF'; 
export const colXs5 = 'col-xs-5--1k3Jf'; 
export const colXs6 = 'col-xs-6--QTdUE'; 
export const colXs7 = 'col-xs-7--3pfc0'; 
export const colXs8 = 'col-xs-8--3BlJ9'; 
export const colXs9 = 'col-xs-9--31s_z'; 
export const colXs10 = 'col-xs-10--4x41f'; 
export const colXs11 = 'col-xs-11--1gr4u'; 
export const colXs12 = 'col-xs-12--2Ml7e'; 
export const colSm1 = 'col-sm-1--9yWJZ'; 
export const colSm2 = 'col-sm-2--2Fskd'; 
export const colSm3 = 'col-sm-3--OMLMx'; 
export const colSm4 = 'col-sm-4--jO_5h'; 
export const colSm5 = 'col-sm-5--2dgyO'; 
export const colSm6 = 'col-sm-6--3dBgF'; 
export const colSm7 = 'col-sm-7--1lHJO'; 
export const colSm8 = 'col-sm-8--2Q_av'; 
export const colSm9 = 'col-sm-9--2aOLm'; 
export const colSm10 = 'col-sm-10--1-_OQ'; 
export const colSm11 = 'col-sm-11--2G-5z'; 
export const colSm12 = 'col-sm-12--SIj9v'; 
export const colMd1 = 'col-md-1--2Qh9d'; 
export const colMd2 = 'col-md-2--2fwrw'; 
export const colMd3 = 'col-md-3--eg5c4'; 
export const colMd4 = 'col-md-4--1L-B0'; 
export const colMd5 = 'col-md-5--3eI4J'; 
export const colMd6 = 'col-md-6--NvxjI'; 
export const colMd7 = 'col-md-7--1oYXn'; 
export const colMd8 = 'col-md-8--3rzJG'; 
export const colMd9 = 'col-md-9--26GAr'; 
export const colMd10 = 'col-md-10--PjqGq'; 
export const colMd11 = 'col-md-11--2YGR1'; 
export const colMd12 = 'col-md-12--Xz8zu'; 
export const colLg1 = 'col-lg-1--2eQh_'; 
export const colLg2 = 'col-lg-2--1Uvq1'; 
export const colLg3 = 'col-lg-3--1MEL9'; 
export const colLg4 = 'col-lg-4--2DqE5'; 
export const colLg5 = 'col-lg-5--2wwQz'; 
export const colLg6 = 'col-lg-6--29FI5'; 
export const colLg7 = 'col-lg-7--2g5tF'; 
export const colLg8 = 'col-lg-8--3ZJNO'; 
export const colLg9 = 'col-lg-9--2EI3i'; 
export const colLg10 = 'col-lg-10--3vzqo'; 
export const colLg11 = 'col-lg-11--2Yhs2'; 
export const colLg12 = 'col-lg-12--3bWg0'; 
export const colXl1 = 'col-xl-1--3lGV1'; 
export const colXl2 = 'col-xl-2--3C9W2'; 
export const colXl3 = 'col-xl-3--2gHpT'; 
export const colXl4 = 'col-xl-4--3C6Qj'; 
export const colXl5 = 'col-xl-5--2C4w8'; 
export const colXl6 = 'col-xl-6--1xFTU'; 
export const colXl7 = 'col-xl-7--2nU0-'; 
export const colXl8 = 'col-xl-8--1-tmv'; 
export const colXl9 = 'col-xl-9--1swVI'; 
export const colXl10 = 'col-xl-10--Ki1Uz'; 
export const colXl11 = 'col-xl-11--3KX9_'; 
export const colXl12 = 'col-xl-12--WC47_'; 
export const section = 'section--OkPIq'; 
export const wrapper = 'wrapper--1S342'; 
export const bottomBlock = 'bottom-block--UU2rk'; 
export const socialWrapper = 'social-wrapper--1Eh8-'; 
export const gutter = 'gutter--1RJfL'; 
export const well = 'well--1Sr1Y'; 
export const footer = 'footer--329-w'; 
export const light = 'light--2vBnY'; 
export const copyright = 'copyright--2q_rg'; 
export const mainBlock = 'main-block--G2y7e'; 
export const logo = 'logo--38B_c'; 
export const columnsContainer = 'columns-container--3xdxk'; 
export const navsWrapper = 'navs-wrapper--2UuSo'; 
export const socialIcons = 'social-icons--2tFpP';
