// @guildeducationinc/recess/components/Navigation/NavBar/index.module.css.js
export const popoverMenu = 'popoverMenu--Czwv-'; 
export const dark = 'dark--3Uxgf'; 
export const popoverMenuButtonContainer = 'popoverMenuButtonContainer--30x7L'; 
export const light = 'light--gRCj0'; 
export const popoverMenuItem = 'popoverMenuItem--2ZTTv'; 
export const popoverMenuItemContainer = 'popoverMenuItemContainer--l9Ye1'; 
export const dangerPopoverMenuItem = 'dangerPopoverMenuItem--wfSss'; 
export const disabledItem = 'disabledItem--22zR3'; 
export const btn = 'btn--1sRtv'; 
export const fullWidth = 'fullWidth--2chJW'; 
export const disabled = 'disabled--2BJl9'; 
export const iconButton = 'iconButton--1xBnz'; 
export const outline = 'outline--36EyN'; 
export const defaultButton = 'defaultButton--2oPUu'; 
export const danger = 'danger--1PSKw'; 
export const primary = 'primary--1c33A'; 
export const text = 'text--2x89j'; 
export const inlineLink = 'inlineLink--xEIoM'; 
export const navWrapper = 'navWrapper--1weJU'; 
export const navDropdown = 'navDropdown--2toVM'; 
export const desktopNavBar = 'desktopNavBar--1j7cz'; 
export const stackNavigator = 'stackNavigator--2iyf5'; 
export const mobileNav = 'mobileNav--1kCdj'; 
export const menuButton = 'menuButton--nG-Wk'; 
export const customActionContainer = 'customActionContainer--rYDje'; 
export const mobileActions = 'mobileActions--3MINh'; 
export const userActionTitle = 'userActionTitle--huV9Z'; 
export const mainNavTabs = 'mainNavTabs--jZ__-'; 
export const measure = 'measure--R8mkN'; 
export const hidden = 'hidden--31CWb'; 
export const brand = 'brand--2VzR4'; 
export const mobileMenuWrapper = 'mobileMenuWrapper--TWJmP'; 
export const mobileMenu = 'mobileMenu--f_R7I'; 
export const navDrawer = 'navDrawer--2B-at'; 
export const pageTitleWrapper = 'pageTitleWrapper--2FolF'; 
export const pageTitle = 'pageTitle--1SNf8'; 
export const subnavWrapper = 'subnavWrapper--3Cs6V'; 
export const subNavContainer = 'subNavContainer--1rrpr'; 
export const backButtonWrapper = 'backButtonWrapper--ByKzK'; 
export const backButtonContainer = 'backButtonContainer--3N7zt'; 
export const mobileMenuList = 'mobileMenuList--3aD09';
