// @guildeducationinc/recess/components/Navigation/NavBar/UserActions.js
import * as React from 'react';
import { Button } from '../../Button';
import { PopoverMenu } from '../../PopoverMenu/PopoverMenu';
import * as styles from './index.module.css.js';
import { BaseButton } from '../../Button/BaseButton';
export function UserActions(_a) {
    var loggedIn = _a.loggedIn, onUserActionsClick = _a.onUserActionsClick, loggedInTitle = _a.loggedInTitle, _b = _a.loggedinActions, loggedinActions = _b === void 0 ? [] : _b, _c = _a.primaryTitle, primaryTitle = _c === void 0 ? 'Sign up' : _c, onPrimaryClick = _a.onPrimaryClick, primaryHref = _a.primaryHref, _d = _a.hidePrimaryAction, hidePrimaryAction = _d === void 0 ? false : _d, _e = _a.secondaryTitle, secondaryTitle = _e === void 0 ? 'Sign in' : _e, onSecondaryClick = _a.onSecondaryClick, secondaryHref = _a.secondaryHref, _f = _a.hideSecondaryAction, hideSecondaryAction = _f === void 0 ? false : _f, testid = _a.testid;
    return loggedIn ? (React.createElement(PopoverMenu, { testid: testid, actions: loggedinActions, id: 'user-actions' },
        React.createElement(BaseButton, { onClick: onUserActionsClick, appearance: 'text', iconDirection: 'right', iconName: 'ChevronDown', className: styles.navDropdown }, loggedInTitle))) : (React.createElement(React.Fragment, null,
        hideSecondaryAction || (React.createElement(Button, { testid: 'secondary-logged-out', appearance: 'text', onClick: onSecondaryClick, href: secondaryHref }, secondaryTitle)),
        hidePrimaryAction || (React.createElement(Button, { testid: 'primary-logged-out', appearance: 'primary', onClick: onPrimaryClick, href: primaryHref }, primaryTitle))));
}
UserActions.displayName = 'UserActions';
UserActions.defaultProps = {
    loggedinActions: [],
    primaryTitle: 'Sign up',
    hidePrimaryAction: false,
    secondaryTitle: 'Sign in',
    hideSecondaryAction: false,
};
