// @guildeducationinc/recess/cypress/popovermenu.helpers.js
export var getButton = function ($element) {
    return cy.wrap($element).find("[data-testid='popover-menu']");
};
export var getMenu = function ($popoverMenu) {
    var testid = $popoverMenu.attr('data-testid');
    return cy.get("[data-testid='" + testid + "--popoverMenu']").find('[data-testid="_popover-menu"]');
};
export var isMenuClosed = function ($popoverMenu) {
    var testid = $popoverMenu.attr('data-testid');
    cy.get("[data-testid='" + testid + "--popoverMenu']").should('not.exist');
};
export var toggleMenu = function ($element) {
    cy.wrap($element).click();
    return cy.wrap($element);
};
export var getActionByIndex = function (index) { return function ($popoverMenu) {
    return cy
        .wrap($popoverMenu)
        .then(getMenu)
        .then(function ($list) {
        return cy.wrap($list.children()[index]);
    });
}; };
export var triggerAction = function (actionName) { return function ($popoverMenu) {
    cy.wrap($popoverMenu)
        .then(getMenu)
        .contains(actionName)
        .click();
    return cy.wrap($popoverMenu);
}; };
export var getAction = function (action) { return function ($element) {
    return cy
        .wrap($element)
        .then(getMenu)
        .contains(action);
}; };
