// @guildeducationinc/recess/cypress/select.helpers.js
var getSelectList = function ($trigger) {
    var testid = $trigger.attr('data-testid');
    return cy.get("[data-testid='" + testid + "--selectPanel']").find('[data-testid="_select-menu-list"]');
};
export var getCurrentValue = function ($trigger) {
    var testid = $trigger.attr('data-testid');
    return cy.get("[data-testid='" + testid + "--wrapper']").find("[data-testid='" + testid + "']");
};
/**
 * A cypress helper to get a option based on it's index.
 * This helper yields the found option. It can be chanined with the `toggleOption` method.
 *
 * @example
 * ```ts
 *    cy.get('[data-testid="myTestId]')
 *      .then(getOptionByIndex(2))
 * ```
 */
export var getOptionByIndex = function (index) { return function ($dropdown) {
    return cy
        .wrap($dropdown)
        .then(getSelectList)
        .then(function ($list) {
        return cy.wrap($list.children()[index]);
    });
}; };
/**
 * A cypress helper to get a option based on it's label
 * This helper yields the found option. It can be chanined with the `toggleOption` method.
 *
 * @example
 * ```ts
 *    cy.get('[data-testid="myTestId]')
 *      .then(getOption('Option 1'))
 * ```
 */
export var getOption = function (option) { return function ($dropdown) {
    return cy
        .wrap($dropdown)
        .then(getSelectList)
        .contains('li', option);
}; };
/**
 * A cypress helper to toggle an option as checked or unchecked, based on it's current state
 * This helper yields the toggled option.
 *
 * @example
 * ```ts
 *    cy.get('[data-testid="myTestId]')
 *      .then(toggleOption('Option 1'))
 * ```
 */
export var toggleOption = function (option) { return function ($dropdown) {
    if (Number(option)) {
        getOptionByIndex(option)($dropdown).click();
    }
    else {
        getOption(option)($dropdown).click();
    }
}; };
/**
 * A cypress helper to toggle a dropdown open or closed.
 * This helper yields the dropdown and can be chained with any other method.
 *
 * @example
 * ```ts
 *    cy.get('[data-testid="myTestId]')
 *      .then(toggleDropdown)
 *      .then(toggleOption('Option 2'))
 * ```
 */
export var toggleSelect = function ($selectButton) {
    cy.wrap($selectButton).click();
    return cy.wrap($selectButton);
};
export var clickOutside = function ($selectButton) {
    cy.wrap($selectButton.parent()).click();
    return cy.wrap($selectButton);
};
