// @guildeducationinc/recess/components/MultiSelect/index.module.css.js
export const boxShadowSm = 'boxShadow--sm--1gzg6'; 
export const dropdownWrapper = 'dropdownWrapper--1qtjF'; 
export const boxShadowMd = 'boxShadow--md--2ukYH'; 
export const boxShadowLg = 'boxShadow--lg--3Bp8J'; 
export const labelWrapper = 'labelWrapper--2eLiE'; 
export const label = 'label--1KPL6'; 
export const disabled = 'disabled--J5HpN'; 
export const invalid = 'invalid--3AqiQ'; 
export const error = 'error--2imrt'; 
export const hidden = 'hidden--1SwIz'; 
export const dropdownTrigger = 'dropdownTrigger--22D1d'; 
export const placeholder = 'placeholder--36PJT'; 
export const hiddenValue = 'hiddenValue--1OE-_'; 
export const open = 'open--2TKZx'; 
export const fadeIn = 'fadeIn--35atY'; 
export const openArrow = 'openArrow--1o4Wv'; 
export const focusedDropdownTrigger = 'focusedDropdownTrigger--6Gy-J'; 
export const listOptions = 'listOptions--2kNkc'; 
export const listOption = 'listOption--3wUdD'; 
export const focusedListOption = 'focusedListOption--yAZo3'; 
export const optionLabel = 'optionLabel--WV6Cl'; 
export const disabledOption = 'disabledOption--3vKbC'; 
export const optionLabelChecked = 'optionLabelChecked--3Aeqd'; 
export const nativeCheckbox = 'nativeCheckbox--F8P3U'; 
export const indeterminateRow = 'indeterminateRow--IIu0K'; 
export const indeterminateWrapper = 'indeterminate-wrapper--2tz60'; 
export const activeFill = 'active-fill--1rLy-'; 
export const subItem = 'sub-item--2LU8I'; 
export const noPanelLabelWrapper = 'noPanelLabelWrapper--1r6yg';
